src/encoded/commands/deploy.py

```diff
@@ -441,119 +294,243 @@ def _wait_and_tag_instances(main_args, run_args, instances_tag_data, instances,
+            print(index, result)
+            is_clean = False
+            break
+    return is_clean
+
+
+def _test_config(title, kwargs):
+    print('\n')
+    print(title)
+    build_config, build_path = _get_cloud_config_yaml(**kwargs)
+    kwargs['build_new_config'] = False
+    prebuilt_config, prebuilt_path = _get_cloud_config_yaml(**kwargs)
+    print('build cc path: {}'.format(build_path))
+    print('build cc path: {}'.format(prebuilt_path))
+    print('Diff: ')
+    return _diff_configs(build_config, prebuilt_config)
+
+
+def _test_cloud_configs(branch):
+    import copy
+
+    kwargs = {
+        'branch': branch,
+        'build_new_config': True,
+        'cluster_name': 'dryruncc-cluster',
+        'conf_dir': './cloud-config',
+        'elasticsearch': None,
+        'no_es': False,
+        'single_data_master': False,
+        'use_local_config': False,
+    }
+
+    # All systems on one machine, Current development demo
+    # cloud-config.yml -> demo.yml
+    title = 'Testing Demo with on board ES'
+    demo_kwargs = copy.copy(kwargs)
+    if not _test_config(title, demo_kwargs):
+        return False
+
+    title = 'Testing Prod/Test/RC/Clustered-Demo frontend with separate ES'
+    frontend_kwargs = copy.copy(kwargs)
+    frontend_kwargs.update({'no_es': True})
+    frontend_kwargs.update({'cluster_name': 'dryruncc-cluster'})
+    if not _test_config(title, frontend_kwargs):
+        return False
+
+    # ES data node cluster with one node elected as master
+    # cloud-config-elasticsearch.yml -> es-elect-head.yml
+    title = 'Testing ES cluster that self elects a head node'
+    es_elect_head_kwargs = copy.copy(kwargs)
+    es_elect_head_kwargs.update({
+        'cluster_name': 'dryruncc-cluster',
+        'elasticsearch': 'yes',
+    })
+    if not _test_config(title, es_elect_head_kwargs):
+        return False
+
+    # ES data node cluster that needs a es head node
+    # cloud-config-elasticsearch.yml -> es-wait-head.yml
+    title = 'Testing ES cluster that wait for a head node to be built'
+    es_wait_head_kwargs = copy.copy(kwargs)
+    es_wait_head_kwargs.update({
+        'cluster_name': 'dryruncc-cluster',
+        'elasticsearch': 'yes',
+        'single_data_master': True,
+    })
+    if not _test_config(title, es_wait_head_kwargs):
+        return False
+
+    # ES head data node for data node cluster
+    # cloud-config-elasticsearch.yml -> es-head.yml
+    title = 'Testing ES cluster head node, required for wait es cluster'
+    es_head_kwargs = copy.copy(kwargs)
+    es_head_kwargs.update({
+        'cluster_name': 'dryruncc-cluster',
+        'single_data_master': True,
+    })
+    if not _test_config(title, es_head_kwargs):
+        return False
+    return True
+
+
 def main():
     # Gather Info
     main_args = parse_args()
+    if main_args.dry_run_cc:
+        # Testing Only
+        if _test_cloud_configs(main_args.branch):
+            sys.exit(0)
+        sys.exit(5)
```

Comment:
oooh exit code 5... fancy.  Not sure it has meaning to anyone though.

Reply:
nope, I just picked different numbers for different failures.